--- /var/db/repos/gentoo/sys-libs/glibc/glibc-2.37-r7.ebuild	2023-12-26 18:42:43.222584593 +0000
+++ /var/db/repos/jhatler-overlay/sys-libs/glibc/glibc-2.37-r7.ebuild	2023-12-31 03:45:16.228405205 +0000
@@ -163,6 +163,10 @@
 	# https://sourceware.org/PR25909
 	tst-support_descriptors
 
+	# tests fail due to hle not being detected in cpuinfo
+	tst-cpu-features-cpuinfo
+	tst-cpu-features-cpuinfo-static
+
 	# The following tests fail only inside portage
 	# https://bugs.gentoo.org/831267
 	tst-system
