--- /var/db/repos/gentoo/x11-base/xorg-proto/xorg-proto-2023.2.ebuild	2023-12-26 18:42:43.319251264 +0000
+++ /var/db/repos/jhatler-overlay/x11-base/xorg-proto/xorg-proto-2023.2.ebuild	2024-01-07 08:57:45.768395370 +0000
@@ -2,7 +2,7 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
 
 MY_PN="${PN/xorg-/xorg}"
 MY_P="${MY_PN}-${PV}"
