--- /var/db/repos/gentoo/dev-python/send2trash/send2trash-1.8.2.ebuild	2023-12-26 18:42:41.952584521 +0000
+++ /var/db/repos/jhatler-overlay/dev-python/send2trash/send2trash-1.8.2.ebuild	2023-12-24 05:17:12.490188494 +0000
@@ -21,3 +21,8 @@
 KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86"
 
 distutils_enable_tests pytest
+
+distutils-r1_src_prepare() {
+	eapply --binary "${FILESDIR}/skip-mac-and-windows.patch"
+	eapply_user
+}
