--- /var/db/repos/gentoo/sys-devel/lld/lld-17.0.6.ebuild	2024-01-19 06:20:58.793415108 +0000
+++ /var/db/repos/jhatler-overlay/sys-devel/lld/lld-17.0.6.ebuild	2024-01-22 20:51:22.230874039 +0000
@@ -12,7 +12,7 @@
 LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA"
 SLOT="${LLVM_MAJOR}/${LLVM_SOABI}"
 KEYWORDS="amd64 ~arm arm64 ~loong ~ppc ~ppc64 ~riscv x86"
-IUSE="debug test zstd"
+IUSE="debug lto test zstd"
 RESTRICT="!test? ( test )"
 
 DEPEND="
@@ -67,10 +67,11 @@
 		-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}"
 		-DBUILD_SHARED_LIBS=ON
 		-DLLVM_INCLUDE_TESTS=$(usex test)
-		-DLLVM_ENABLE_ZLIB=FORCE_ON
-		-DLLVM_ENABLE_ZSTD=$(usex zstd FORCE_ON OFF)
+		-DLLVM_ENABLE_ZSTD=$(usex zstd)
 	)
 
+	use lto && mycmakeargs+=( -DLLVM_ENABLE_LTO="Thin" )
+
 	use test && mycmakeargs+=(
 		-DLLVM_EXTERNAL_LIT="${EPREFIX}/usr/bin/lit"
 		-DLLVM_LIT_ARGS="$(get_lit_flags)"
