--- /var/db/repos/gentoo/dev-python/terminado/terminado-0.18.0.ebuild	2023-12-26 18:42:41.965917855 +0000
+++ /var/db/repos/jhatler-overlay/dev-python/terminado/terminado-0.18.0.ebuild	2023-12-24 05:33:36.753577015 +0000
@@ -23,6 +23,12 @@
 	dev-python/tornado[${PYTHON_USEDEP}]
 "
 
+BDEPEND="
+	test? (
+		dev-python/pytest-timeout[${PYTHON_USEDEP}]
+	)
+"
+
 distutils_enable_tests pytest
 
 src_test() {
