--- /var/db/repos/gentoo/dev-python/pytest-virtualenv/pytest-virtualenv-1.7.0-r2.ebuild	2023-12-26 18:42:41.939251187 +0000
+++ /var/db/repos/jhatler-overlay/dev-python/pytest-virtualenv/pytest-virtualenv-1.7.0-r2.ebuild	2023-12-24 07:04:34.600550074 +0000
@@ -37,7 +37,7 @@
 
 python_test() {
 	local EPYTEST_DESELECT=()
-	if [[ ${EPYTHON} == python3.12 ]]; then
+	if [[ ${EPYTHON} == python3.12 ]] || [[ ${EPYTHON} == python3.11 ]]; then
 		EPYTEST_DESELECT+=(
 			# assumes setuptools is installed inside virtualenv
 			# https://github.com/man-group/pytest-plugins/issues/220
