--- /var/db/repos/gentoo/media-libs/jbig2dec/jbig2dec-0.19.ebuild	2023-12-26 18:42:42.422584548 +0000
+++ /var/db/repos/jhatler-overlay/media-libs/jbig2dec/jbig2dec-0.19.ebuild	2024-01-22 21:22:48.947646495 +0000
@@ -3,7 +3,7 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
 inherit autotools python-any-r1
 
 DESCRIPTION="A decoder implementation of the JBIG2 image compression format"
