--- /var/db/repos/gentoo/sys-devel/llvm/llvm-17.0.6.ebuild	2024-01-19 06:20:58.796748441 +0000
+++ /var/db/repos/jhatler-overlay/sys-devel/llvm/llvm-17.0.6.ebuild	2024-01-22 20:51:22.230874039 +0000
@@ -23,6 +23,7 @@
 IUSE="
 	+binutils-plugin debug debuginfod doc exegesis libedit +libffi
 	ncurses test xar xml z3 zstd
+	+polly lto
 "
 RESTRICT="!test? ( test )"
 
@@ -48,6 +49,7 @@
 BDEPEND="
 	${PYTHON_DEPS}
 	dev-lang/perl
+	>=dev-build/cmake-3.16
 	sys-devel/gnuconfig
 	kernel_Darwin? (
 		<sys-libs/libcxx-${LLVM_VERSION}.9999
@@ -131,6 +133,10 @@
 					# TableGen lib + deps
 					LLVMDemangle|LLVMSupport|LLVMTableGen)
 						;;
+					# Polly + deps
+					Polly|LLVMAggressiveInstCombine|LLVMAnalysis|LLVMAsmParser|LLVMAsmPrinter|LLVMBinaryFormat|LLVMBitReader|LLVMBitWriter|LLVMBitstreamReader|LLVMCodeGen|LLVMCore|LLVMCoroutines|LLVMDebugInfoCodeView|LLVMDebugInfoDWARF|LLVMDebugInfoMSF|LLVMDebugInfoPDB|LLVMExtensions|LLVMFrontendOpenMP|LLVMIRPrinter|LLVMIRReader|LLVMInstCombine|LLVMInstrumentation|LLVMLinker|LLVMMC|LLVMMCParser|LLVMNVPTXCodeGen|LLVMNVPTXDesc|LLVMNVPTXInfo|LLVMObjCARCOpts|LLVMObject|LLVMPasses|LLVMProfileData|LLVMRemarks|LLVMScalarOpts|LLVMSelectionDAG|LLVMSymbolize|LLVMTarget|LLVMTargetParser|LLVMTextAPI|LLVMTransformUtils|LLVMVectorize|LLVMipo)
+						use polly || continue
+						;;
 					# testing libraries
 					LLVMTestingAnnotations|LLVMTestingSupport)
 						;;
@@ -177,6 +183,10 @@
 	fi
 }
 
+pre_src_unpack() {
+	use polly && LLVM_COMPONENTS+=( polly )
+}
+
 src_prepare() {
 	# disable use of SDK on OSX, bug #568758
 	sed -i -e 's/xcrun/false/' utils/lit/lit/util.py || die
@@ -330,6 +340,52 @@
 		use binutils-plugin && out+=(
 			LLVMgold
 		)
+		use polly && out+=(
+			Polly
+			LLVMAggressiveInstCombine
+			LLVMAnalysis
+			LLVMAsmParser
+			LLVMAsmPrinter
+			LLVMBinaryFormat
+			LLVMBitReader
+			LLVMBitWriter
+			LLVMBitstreamReader
+			LLVMCodeGen
+			LLVMCodeGenTypes
+			LLVMCore
+			LLVMCoroutines
+			LLVMDebugInfoBTF
+			LLVMDebugInfoCodeView
+			LLVMDebugInfoDWARF
+			LLVMDebugInfoMSF
+			LLVMDebugInfoPDB
+			LLVMExtensions
+			LLVMFrontendOpenMP
+			LLVMIRPrinter
+			LLVMIRReader
+			LLVMInstCombine
+			LLVMInstrumentation
+			LLVMLinker
+			LLVMMC
+			LLVMMCParser
+			LLVMNVPTXCodeGen
+			LLVMNVPTXDesc
+			LLVMNVPTXInfo
+			LLVMObjCARCOpts
+			LLVMObject
+			LLVMPasses
+			LLVMProfileData
+			LLVMRemarks
+			LLVMScalarOpts
+			LLVMSelectionDAG
+			LLVMSymbolize
+			LLVMTarget
+			LLVMTargetParser
+			LLVMTextAPI
+			LLVMTransformUtils
+			LLVMVectorize
+			LLVMipo
+		)
 		use debuginfod && out+=(
 			llvm-debuginfod
 		)
@@ -376,8 +432,7 @@
 		-DLLVM_ENABLE_EH=ON
 		-DLLVM_ENABLE_RTTI=ON
 		-DLLVM_ENABLE_Z3_SOLVER=$(usex z3)
-		-DLLVM_ENABLE_ZLIB=FORCE_ON
-		-DLLVM_ENABLE_ZSTD=$(usex zstd FORCE_ON OFF)
+		-DLLVM_ENABLE_ZSTD=$(usex zstd)
 		-DLLVM_ENABLE_CURL=$(usex debuginfod)
 		-DLLVM_ENABLE_HTTPLIB=$(usex debuginfod)
 
@@ -394,6 +449,15 @@
 		-DOCAMLFIND=NO
 	)
 
+	use polly && mycmakeargs+=(
+		-DLLVM_ENABLE_PROJECTS="polly"
+		-DPOLLY_ENABLE_GPGPU_CODEGEN=ON
+		-DLLVM_TOOL_POLLY_BUILD=ON
+		-DLLVM_POLLY_LINK_INTO_TOOLS=ON
+	)
+
+	use lto && mycmakeargs+=( -DLLVM_ENABLE_LTO="Thin" )
+
 	local suffix=
 	if [[ -n ${EGIT_VERSION} && ${EGIT_BRANCH} != release/* ]]; then
 		# the ABI of the main branch is not stable, so let's include
@@ -440,13 +504,6 @@
 		)
 	fi
 
-	# On Macos prefix, Gentoo doesn't split sys-libs/ncurses to libtinfo and
-	# libncurses, but llvm tries to use libtinfo before libncurses, and ends up
-	# using libtinfo (actually, libncurses.dylib) from system instead of prefix
-	use kernel_Darwin && mycmakeargs+=(
-		-DTerminfo_LIBRARIES=-lncurses
-	)
-
 	# workaround BMI bug in gcc-7 (fixed in 7.4)
 	# https://bugs.gentoo.org/649880
 	# apply only to x86, https://bugs.gentoo.org/650506
@@ -510,6 +567,10 @@
 multilib_src_install() {
 	DESTDIR=${D} cmake_build install-distribution
 
+	if use polly; then
+		DESTDIR=${D} cmake_build tools/polly/install/strip
+	fi
+
 	# move headers to /usr/include for wrapping
 	rm -rf "${ED}"/usr/include || die
 	mv "${ED}"/usr/lib/llvm/${LLVM_MAJOR}/include "${ED}"/usr/include || die
