--- /var/db/repos/gentoo/media-libs/graphene/graphene-1.10.8.ebuild	2023-12-26 18:42:42.419251214 +0000
+++ /var/db/repos/jhatler-overlay/media-libs/graphene/graphene-1.10.8.ebuild	2024-01-10 23:20:28.365380623 +0000
@@ -3,7 +3,7 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
 inherit xdg-utils meson-multilib python-any-r1
 
 DESCRIPTION="A thin layer of types for graphic libraries"
