--- /var/db/repos/gentoo/net-misc/rsync/rsync-3.2.7-r4.ebuild	2024-01-06 19:48:47.427321639 +0000
+++ /var/db/repos/jhatler-overlay/net-misc/rsync/rsync-3.2.7-r4.ebuild	2024-01-07 08:57:45.765062036 +0000
@@ -5,7 +5,7 @@
 
 # Uncomment when introducing a patch which touches configure
 RSYNC_NEEDS_AUTOCONF=1
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
 inherit flag-o-matic prefix python-single-r1 systemd
 
 DESCRIPTION="File transfer program to keep remote files into sync"
